src/org/opensolaris/opengrok/web/Statistics.java

```diff
@@ -103,11 +126,11 @@ protected void maybeRefresh() {
      * @param category category
      */
     public void addRequest(HttpServletRequest req, String category) {
-        Long val = requestCategories.get(category);
+        AtomicLong val = requestCategories.get(category);
         if (val == null) {
-            val = new Long(0);
+            val = new AtomicLong(0);
```

Comment:
Not sure this would work. If 2 threads are racing for this code while `val` is still `null`, both can read `val` as `null` so both will create new instance and 1 count will be lost.